rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java

```diff
@@ -73,28 +73,32 @@ static List<Path> runtimeClasspath() {
      * matching jars can be found.
      */
     static List<Path> dependenciesFromClasspath(String... artifactNames) {
-        Map<String, Pattern> artifactNamePatterns = Arrays.stream(artifactNames)
-                .collect(toMap(Function.identity(), name -> Pattern.compile(name + "-.*?\\.jar$")));
-
-        Set<String> foundArtifacts = new HashSet<>();
-        List<Path> artifacts = new ArrayList<>();
         List<URI> runtimeClasspath = new ClassGraph().getClasspathURIs();
-        for (URI cpEntry : runtimeClasspath) {
-            for (Map.Entry<String, Pattern> artifactNamePattern : artifactNamePatterns.entrySet()) {
```

Comment:
I turned these 2 loops inside out, as that is more performant (the inner loop can be shortcutted with break) and uses less memory (no need to create a map of Patterns).